lib/scrub/pre_load_backup.rb

```diff
@@ -0,0 +1,83 @@
+require "json"
+
+# Use this class in the following context:
+# You want to load a new monographs file from Test U.
+# First you want to make sure that the already loaded records are backed up,
+# and then deleted.
+#
+# Scrub::PreLoadBackup.new(...).run will write the backup file and mark
+# the existing matching records for deletion.
+# Scrub::PreLoadBackup.new(...).delete_marked! will delete matching marked records.
+#
+# Invoke via phctl:
+# $ bash phctl.sh backup holdings --organization umich --mono_multi_serial spm
+# $ bash phctl.sh backup holdings --organization umich --mono_multi_serial spm mpm ser
+
+module Scrub
+  class PreLoadBackup
+    attr_reader :organization, :mono_multi_serial
+
+    def initialize(organization:, mono_multi_serial:)
+      @organization = organization
+      @mono_multi_serial = mono_multi_serial
+      validate
+    end
+
+    def write_backup_file
+      # Note that this writes an empty file if there are no matching records.
+      File.open(backup_path, "w") do |backup_file|
+        records.each do |record|
```

Comment:
We should use `paged_each` as in `Clusterable::Holding` to avoid waiting for the entire query to complete and (I think) loading everything into memory.

Reply:
Ah, yes. Added `paged_each`.